src/Core/System/Snippet/Struct/ValidatedTranslationFileOptions.php

```diff
@@ -0,0 +1,76 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Snippet\Struct;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Struct\Struct;
+use Shopware\Core\System\Snippet\Command\ValidateTranslationFilesCommand;
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * @internal
+ */
+#[Package('discovery')]
+class ValidatedTranslationFileOptions extends Struct
+{
+    /**
+     * @param list<string> $extensionPaths
+     * @param list<string> $ignoredPaths
+     */
+    public function __construct(
+        protected bool $fix = false,
```

Comment:
also here readonly public properties

Reply:
Why has every struct protected properties then? Just to know for the future

Reply:
https://github.com/shopware/shopware/pull/12188#discussion_r2379468413

they are protected so they are considered during JSON encoding. but they are not public properties because they are older than the PHP version, which allows such modern writing of DTOs :slightly_smiling_face: